### DISCUSSION TITLE
AI Solution

### DISCUSSION THREAD

**Discussion by matteomeli** [o3de/o3de#1961] [General]
Hello, I have looked around but haven't found any question yet about AI.

Is there going to be any AI "gem" available withing the engine?

I have noticed, that [Kythera AI](https://www.kythera.ai/) claims to be "now the AI solution for the Open 3D Engine!" (from a blog [post](https://www.kythera.ai/news/kythera-ai-for-o3de-were-part-of-a-new-open-source-game-engine) on their website).

In another [page ](https://www.kythera.ai/licensing) in their website, under licensing, they also specify regarding the license for O3DE, that:

**Non-commercial Licences (O3DE)**

_We are working to give back to the games development community. Our first platform for this is the O3DE engine, where Kythera AI is available to download under the terms of our Community License. It allows commercial use based on a royalty, with a buyout option - and it’s also free for non-commercial use._

Furthermore, their source code is not compatible with an Apache 2.0 open source licensing, as it's stated in their [EULA page](https://www.kythera.ai/o3de-eula):

_As such, the set of compilable programming language text files such as .h and .cpp files (“Source Code”) included with the Software is not under an open source license and you are not allowed to redistribute those files, except where clearly marked as such. You are entitled under this EULA to use the Software with the O3DE engine in relation to your video computer game projects or any other project.  There is no royalty where such use is non-commercial. Where such use involves commercial application or distribution then royalties will be due, as explained in the commercial terms mentioned above._

How is this compatible with the strong open-source nature of the engine?

**Comment by AreebDurrani** [comment]
The LmbrCentral gem already provides navmesh functionality separate from Kythera. I have created a discussion in the rfcs project about expanding it with new features such as behavior trees and spatial queries.
